src/Ocean/Domains/rectangular_domain.jl

```diff
@@ -0,0 +1,194 @@
+a `RectangularDomain`.
+"""
+function RectangularDomain(grid::DiscontinuousSpectralElementGrid, Ne)
+    Ne = name_it(Ne)
+
+    # Unwind volume geometry
+    volume_geometry = grid.vgeo
+
+    # Check number of elements
+    prod(Ne) === size(volume_geometry, 3) ||
+    error("prod(Ne) must match the total number of grid elements.")
+
+    Np = polynomialorder(grid)
+
+    x = view(volume_geometry, :, grid.x1id, :)
+    y = view(volume_geometry, :, grid.x2id, :)
+    z = view(volume_geometry, :, grid.x3id, :)
+
+    xlims = (minimum(x), maximum(x))
+    ylims = (minimum(y), maximum(y))
+    zlims = (minimum(z), maximum(z))
+
+    L = (
+        x = xlims[2] - xlims[1],
+        y = ylims[2] - ylims[1],
+        z = zlims[2] - zlims[1],
+    )
+
+    return RectangularDomain(grid, Np, Ne, L, xlims, ylims, zlims)
+end
+
+"""
+    RectangularDomain(FT=Float64;
+                      elements,
+                      polynomialorder,
+                      x = (-1, 1),
+                      y = (-1, 1),
+                      z = (-1, 1),
+                      periodicity = (true, true, false),
+                      boundary = ((0, 0), (0, 0), (1, 2)),
+                      array_type = Array,
+                      message_communicator = MPI.COMM_WORLD)
+
+Returns a `RectangularDomain` representing the product of `x, y, z` intervals,
+specified by 2-tuples.
+
+The `RectangularDomain` is meshed with a simple `DiscontinuousSpectralElementGrid`
+with an isotropic `polynomialorder` and a 3-tuple of `elements`
+giving the number of elements in `x, y, z`.
+
+Additional arguments are:
+
+- `periodicity`: a 3-tuple that indicates periodic dimensions with `true`, 
+
+- `boundary`: specifies the boundary condition on each boundary with a
+              boundary condition `tag`
+
+- `array_type`: either `Array` for CPU computations or `CuArray` for
+                GPU computations
+
+- `message_communicator`: the world communicator for message passing between
+                          nodes in a distributed memory configuration.
+
+Example
+=======
+
+```jldoctest
+julia> using ClimateMachine; ClimateMachine.init()
+
+julia> using ClimateMachine.Ocean.RectangularDomains: RectangularDomain
+
+julia> domain = RectangularDomain(elements=(7, 8, 9), polynomialorder=4, x=(0, 1), y=(0, 1), z=(0, 1))
+RectangularDomain{Float64, ClimateMachine.Mesh.Grids.DiscontinuousSpectralElementGrid}:
+    Np = 4, Ne = (x = 7, y = 8, z = 9)
+    L = (x = 1.00e+00, y = 1.00e+00, z = 1.00e+00)
+    x = (0.00e+00, 1.00e+00), y = (0.00e+00, 1.00e+00), z = (1.00e+00, 0.00e+00)
+```
+"""
+function RectangularDomain(
+    FT = Float64;
+    elements,
+    polynomialorder,
+    x::Tuple{<:Number, <:Number},
+    y::Tuple{<:Number, <:Number},
+    z::Tuple{<:Number, <:Number},
+    periodicity = (true, true, false),
+    boundary = ((0, 0), (0, 0), (1, 2)),
+    array_type = Array,
+    message_communicator = MPI.COMM_WORLD,
+)
```

Comment:
```suggestion
function RectangularDomain(
    ::Type{FT};
    elements,
    polynomialorder,
    x::Tuple{<:Number, <:Number},
    y::Tuple{<:Number, <:Number},
    z::Tuple{<:Number, <:Number},
    periodicity = (true, true, false),
    boundary = ((0, 0), (0, 0), (1, 2)),
    array_type = Array{FT},
    message_communicator = MPI.COMM_WORLD,
) where {FT}
```
Or, we could grab `FT` from `eltype` of `array_type`?

Reply:
That seems like a good idea --- more compact specification is always nice. And just to be clear, we use `array_type` to distinguish between CPU / GPU computations?

Reply:
Yes. The array type is a [global setting](https://github.com/CliMA/ClimateMachine.jl/blob/master/src/Driver/Driver.jl#L92) -- the default should come from there [like so](https://github.com/CliMA/ClimateMachine.jl/blob/master/src/Driver/driver_configs.jl#L123).

However, there is no global setting for the floating point precision selection (but there should be) so we require that it [be specified](https://github.com/CliMA/ClimateMachine.jl/blob/master/src/Driver/driver_configs.jl#L140).

Reply:
Hmm.

Thinking about it more, I think it's cumbersome to have to write `Array{Float64}` rather than `Array`. The vast majority of users will use double precision; it's only specialty applications that need to changing the floating point type (that we know of). Thus I think it makes sense to specify the floating point precision and the array type with independent kwargs.

I'll configure this constructor to use the global setting as a default.